lib/connections/connections.go

```diff
@@ -35,17 +37,39 @@ var (
 	listeners = make(map[string]ListenerFactory, 0)
 )
 
+type ConnectionHandlingModel interface {
```

Comment:
Renamed to Model.

This interface needs the full protocol.Model interface, because the object is passed to protocol.NewConnection, which expects a protocol.Model.

I couldn't find anything in Go that would allow for interface extension, so I duplicated the methods here. They are all required for compilation to succeed, though.